src/WebJobs.Script/Scale/HostPerformanceManager.cs

```diff
@@ -27,20 +39,99 @@ public HostPerformanceManager(IEnvironment environment, IOptions<HostHealthMonit
+        {
+            var hostProcessStats = _processMonitor.GetStats();
+            if (hostProcessStats.CpuLoadHistory.Any())
+            {
+                string formattedLoadHistory = string.Join(",", hostProcessStats.CpuLoadHistory);
+                logger?.LogDebug($"[HostMonitor] Host process stats: EffectiveCores={_environment.GetEffectiveCoresCount()}, CpuLoadHistory=({formattedLoadHistory}), AvgLoad: {Math.Round(hostProcessStats.CpuLoadHistory.Average())}, MaxLoad: {Math.Round(hostProcessStats.CpuLoadHistory.Max())}");
+            }
+
+            double workersCpuAverage = 0;
+            var dispatcher = GetDispatcherAsync();
+            if (dispatcher != null)
+            {
+                // If OOP, check worker stats
+                var workerStatuses = await dispatcher.GetWorkerStatusesAsync();
+                if (workerStatuses.All(p => p.Value.ProcessStats.CpuLoadHistory.Count() > MinSampleCount))
+                {
+                    // first compute the average CPU load for each worker
+                    var averageWorkerCpuStats = new List<double>();
+                    foreach (var currentStatus in workerStatuses)
+                    {
+                        var workerProcessStats = currentStatus.Value.ProcessStats;
+                        int currWorkerProcessCpuStatsCount = workerProcessStats.CpuLoadHistory.Count();
+                        if (currWorkerProcessCpuStatsCount >= MinSampleCount)
+                        {
+                            // take the last 5 samples
+                            var currWorkerCpuStats = workerProcessStats.CpuLoadHistory.Skip(currWorkerProcessCpuStatsCount - MinSampleCount).Take(MinSampleCount).Average();
+                            averageWorkerCpuStats.Add(currWorkerCpuStats);
+                        }
+                    }
+
+                    // compute the final average across all workers
+                    workersCpuAverage = averageWorkerCpuStats.Average();
+                }
+            }
+
+            // calculate the aggregate load of host + workers (if OOP)
+            int hostProcessCpuStatsCount = hostProcessStats.CpuLoadHistory.Count();
+            if (hostProcessCpuStatsCount > MinSampleCount)
+            {
+                // compute the aggregate average for host + workers for the last MinSampleCount samples
+                var hostProcessCpuStatsAverage = hostProcessStats.CpuLoadHistory.Skip(hostProcessCpuStatsCount - MinSampleCount).Take(MinSampleCount).Average();
+                var aggregateAverage = Math.Round(hostProcessCpuStatsAverage + workersCpuAverage);
+                logger?.LogDebug($"[HostMonitor] Host aggregate load {aggregateAverage}");
+
+                // if the average is above our threshold, return true (we're overloaded)
+                var adjustedThreshold = _healthMonitorOptions.Value.CounterThreshold * 100;
```

Comment:
Chose to use the existing configurable CounterThreshold here which defaults to 80%. While it does make this configurable w/o adding a new config knob, it does mean that the existing knob will also participate in scale behavior.